File: go/base/commons/displayable.go
package commons

import "fmt"

type Displayable struct{}

func (Displayable) Displayable() bool {
	return true
}

func (d *Displayable) Display() {
	if d.Displayable() {
		fmt.Println("Yes, this shape can be displayed.")
	} else {
		fmt.Println("No, this shape cannot be displayed.")
	}
}

File: cplusplus/src/base/commons/displayable.cpp
#include <print>
#include <shapes/base/commons.hpp>

namespace shapes::base::commons {
Displayable::~Displayable() = default;

auto Displayable::shouldDisplay() const -> bool {
    return true;
}

void Displayable::display() const {
    if (shouldDisplay()) {
        std::println("Yes, this shape can be displayed.");
    } else {
        std::println("No, this shape cannot be displayed.");
    }
}
} // namespace shapes::base::commons
